### PR TITLE
Add a unittest for the EmailStep.

### PR DESCRIPTION
**Pull request by picklesrus**
### Test Coverage:
yep!

### CODE REVIEW

File: test/unit/components/email-step.test.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {shallowWithIntl} from '../../helpers/intl-helpers.jsx';
+import EmailStep from '../../../src/components/join-flow/email-step.jsx';
+import JoinFlowStep from '../../../src/components/join-flow/join-flow-step.jsx';
```
Comment (rschamp) [inline test/unit/components/email-step.test.jsx]: This file uses a mix of `import` and `require`. I think `require` is the broadly used way in this repo, so we should consolidate on that.

Comment (picklesrus) [reply test/unit/components/email-step.test.jsx]: Good call. done!

**Comment by picklesrus** [comment]
Fixed the import/require. I also changed the initial data in one of the objects to be less cute since it was confusing on initial reading of the code. Lastly, Ben pointed out that there were better ways to do the object equality check for arguments to toHaveBeenCalledWith so I changed that too.

**Comment by rschamp** [comment]
Looks good! I agree the updated equality checks look better. Thanks!

**Pull request merged**